### DISCUSSION TITLE
Hackweek Work

### DISCUSSION THREAD

**Discussion by briandowns** [Jason2605/Dictu#582] [Q&A]
We're going to be doing a hackweek at my work soon and I was hoping to get your (@Jason2605) opinion on somethings that might be good to get working, added, or expanded.

A few ideas I had were:

* RPM builds for new releases
* Addition to the FreeBSD Ports tree
* #575 items

**Comment by Jason2605** [comment]
Hey! Sounds cool! Yeah sorting out builds for the release with Github Actions would be cool! Anything to do with the VSc plugin is a bonus, it's in a pretty poor state at the moment. The documentation could probably do with a bit of a revamp as well - making sure examples and text is up to date along with adding return types throughout.

A big one that needs to be done is adding fibers #487, [Wren](https://github.com/wren-lang/wren) is very similar to Dictu internally so there is a lot of code that we can re-use from that repository to get some sort of concurrency into Dictu - this is the main things that's holding me off a 1.x.x release (and a couple of the bugs found ofc).

There are a couple of other big ticket items in the issues, notably Unicode support #317 and method annotations #464. 

A cool project that's not strictly related to the Dictu VM itself would be a Dictu web playground #262 (unsure if this is just you who's looking to contribute during the workweek, but if there are other potential contributors who don't care about learning the C codebase, and know some web programming that's an option for them?). Essentially something we could tack onto the docs site (probably on a subdomain) so people can give it a quick play without having to download everything initially would be cool.

I'm sure there are more things that'll come to me in the near future and I'll jot them down here
